### PR TITLE
Refactor hide layer button

### PR DESCRIPTION
**Pull request by junqiu-lei**
### Description
Refactor hide layer button from layerControlPanel into individual component and added test.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/layer_control_panel/hide_layer_button.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  LAYER_VISIBILITY,
+} from '../../../common';
+import { MapLayerSpecification } from '../../model/mapLayerType';
+import { updateLayerVisibility } from '../../model/map/layer_operations';
+
+interface MaplibreRef {
+  current: Maplibre | null;
+}
+
+interface DeleteLayerButtonProps {
```
Comment (VijayanB) [inline public/components/layer_control_panel/hide_layer_button.tsx:21]:
```suggestion
interface HideLayerButtonProps {
```

File: public/components/layer_control_panel/hide_layer_button.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  layer: MapLayerSpecification;
+  maplibreRef: MaplibreRef;
+}
+export const HideLayerButton = ({ layer, maplibreRef }: DeleteLayerButtonProps) => {
```
Comment (VijayanB) [inline public/components/layer_control_panel/hide_layer_button.tsx:25]:
```suggestion
export const HideLayer = ({ layer, maplibreRef }: DeleteLayerButtonProps) => {
```

File: public/components/layer_control_panel/hide_layer_button.tsx (hunk 1 above)
Comment (junqiu-lei) [reply public/components/layer_control_panel/hide_layer_button.tsx]: updated

File: public/components/layer_control_panel/hide_layer_button.tsx (hunk 2 above)
Comment (junqiu-lei) [reply public/components/layer_control_panel/hide_layer_button.tsx]: updated

File: public/components/layer_control_panel/hide_layer_button.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  const onLayerVisibilityChange = (selectedLayer: MapLayerSpecification) => {
+    if (selectedLayer.visibility === LAYER_VISIBILITY.VISIBLE) {
+      selectedLayer.visibility = LAYER_VISIBILITY.NONE;
+      setLayerVisibility(new Map(layerVisibility.set(layer.id, false)));
+    } else {
+      layer.visibility = LAYER_VISIBILITY.VISIBLE;
+      setLayerVisibility(new Map(layerVisibility.set(layer.id, true)));
+    }
+    updateLayerVisibility(maplibreRef.current!, layer.id, layer.visibility);
```
Comment (VijayanB) [inline public/components/layer_control_panel/hide_layer_button.tsx:36]: We shouldn't be modifying props directly. Treat props as read only and pass mutation as function from parent.

Comment (junqiu-lei) [reply public/components/layer_control_panel/hide_layer_button.tsx]: Updated to use updateLayer(updatedLayer) to change the property from parent

File: public/components/layer_control_panel/hide_layer_button.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  maplibreRef: MaplibreRef;
+  updateLayer: (layer: MapLayerSpecification) => void;
+}
+export const HideLayer = ({ layer, maplibreRef, updateLayer }: HideLayerProps) => {
+  const [layerVisibility, setLayerVisibility] = useState(
+    new Map<string, boolean>([[layer.id, layer.visibility === LAYER_VISIBILITY.VISIBLE]])
+  );
+
+  const onLayerVisibilityChange = () => {
+    let updatedLayer: MapLayerSpecification;
```
Comment (VijayanB) [inline public/components/layer_control_panel/hide_layer_button.tsx:32]: create a new object with only visibility instead of entire layer.

Comment (junqiu-lei) [reply public/components/layer_control_panel/hide_layer_button.tsx:32]: We also need to update the existing layer, so we need the layer object to replace the existing:
```
updatedLayer = {
        ...layer,
        visibility: LAYER_VISIBILITY.NONE,
      };
updateLayer(updatedLayer);
```

Comment (junqiu-lei) [reply public/components/layer_control_panel/hide_layer_button.tsx]: Updated to have a function `updateLayerVisibility` to only pass the visibility props

**Comment by junqiu-lei** [comment]
The CI failure is from upstream cluster, not related to this PR

**Review by VijayanB — APPROVED** [review]

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

**Pull request merged**